rkt/pubkey/pubkey.go

```diff
@@ -87,24 +82,22 @@ func (m *Manager) AddKeys(pkls []string, prefix string, accept AcceptOption, ove
 		}
 		defer pk.Close()
 
-		exists, err := m.Ks.TrustedKeyPrefixExists(prefix, pk)
-		if err != nil {
-			return fmt.Errorf("error reading the key %s: %v", pkl, err)
-		}
 		err = displayKey(prefix, pkl, pk)
 		if err != nil {
 			return fmt.Errorf("error displaying the key %s: %v", pkl, err)
 		}
-		if exists && override == OverrideDeny {
-			stderr("Key %q already in the keystore", pkl)
-			continue
-		}
 
 		if m.TrustKeysFromHTTPS && u.Scheme == "https" {
 			accept = AcceptForce
 		}
 
 		if accept == AcceptAsk {
+			if !terminal.IsTerminal(int(os.Stderr.Fd())) {
```

Comment:
I think this should check both `os.Stdin` and `os.Stderr` for now.

-----

Ideally, `reviewKey` should prompt and read on `/dev/tty` rather than stdin/stderr (in the same way as ssh ask for the password on `/dev/tty`). That would allow piping like `echo foo | rkt run image.aci --interactive | grep bar` and still get prompted about the key on the terminal. Then, no need to call `terminal.IsTerminal()`: we would check the error (ENXIO) when opening `/dev/tty` instead. But it is out of scope of this PR. /cc @blixtra @jonboulle we were talking about `stderr()` in `reviewKey()` today.